lambdas/call-ticf-cri/src/main/java/uk/gov/di/ipv/core/callticfcri/service/TicfCriService.java

```diff
@@ -0,0 +1,133 @@
+
+public class TicfCriService {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    public static final String TRUSTMARK = "https://oidc.account.gov.uk/trustmark";
+    public static final String X_API_KEY_HEADER = "x-api-key";
+
+    private final ConfigService configService;
+    private final HttpClient httpClient;
+    private final VerifiableCredentialJwtValidator jwtValidator;
+
+    public TicfCriService(ConfigService configService) {
+        this.configService = configService;
+        this.httpClient = HttpClient.newHttpClient();
+        this.jwtValidator = new VerifiableCredentialJwtValidator(configService);
+    }
+
+    protected TicfCriService(
+            ConfigService configService,
+            HttpClient httpClient,
+            VerifiableCredentialJwtValidator jwtValidator) {
+        this.configService = configService;
+        this.httpClient = httpClient;
+        this.jwtValidator = jwtValidator;
+    }
+
+    public List<SignedJWT> getTicfVc(
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            IpvSessionItem ipvSessionItem,
+            List<String> credentials)
+            throws TicfCriServiceException {
+        try {
+            BackEndCriConfig ticfCriConfig = configService.getBackEndCriConfig(TICF_CRI);
+
+            TicfCriDto ticfCriRequest =
+                    new TicfCriDto(
+                            clientOAuthSessionItem.getVtr(),
+                            ipvSessionItem.getVot(),
+                            TRUSTMARK,
+                            clientOAuthSessionItem.getUserId(),
+                            clientOAuthSessionItem.getGovukSigninJourneyId(),
+                            credentials);
+
+            HttpRequest.Builder httpRequestBuilder =
+                    HttpRequest.newBuilder()
+                            .uri(ticfCriConfig.getCredentialUrl())
+                            .POST(
+                                    HttpRequest.BodyPublishers.ofString(
+                                            objectMapper.writeValueAsString(ticfCriRequest)));
+            if (ticfCriConfig.requiresApiKey()) {
+                httpRequestBuilder.header(
+                        X_API_KEY_HEADER,
+                        configService.getCriPrivateApiKeyForActiveConnection(TICF_CRI));
+            }
+
+            HttpResponse<String> ticfCriHttpResponse;
+            try {
+                ticfCriHttpResponse = sendHttpRequest(httpRequestBuilder.build());
+                checkStatusCode(ticfCriHttpResponse);
+            } catch (IOException
+                    | InterruptedException
+                    | IllegalArgumentException
+                    | SecurityException
+                    | TicfCriServiceException e) {
+                // In the case of unavailability, the TICF CRI is deemed optional.
```

Comment:
Have you checked this with David Page?

I think the idea is that the TICF CRI should _always_ return a response to us, it may just not contain any evidence if the service wasn't able to respond. This allows us to prove that we tried to run the check to SPOT etc.

Reply:
Ok, definitely check this with David. My understanding is that the TICF CRI should always give us a response, and it will handle issues with the underlying risk engine is unresponsive, but he should be able to confirm.